tests/unit/test_util.py

```diff
@@ -140,7 +140,47 @@ def test_apply_negative_precision_threshold_valid_positive_input(self) -> None:
         )
         self.assertEqual(func_out_2, 0.000001)
 
-    def test_jit_test(self):
+    def test_solve_qp_invalid_kernel_mm(self) -> None:
+        """
+        Test how solve_qp handles invalid inputs of kernel_mm.
+
+        The output of solve_qp is indirectly tested when testing the various weight
+        optimisers that are used in this codebase. This test just ensures sensible
+        behaviour occurs when unexpected inputs are passed to the function.
+        """
+        # Attempt to solve a QP with an input that cannot be converted to a JAX array -
+        # this should error as no sensible result can be found in such a case.
+        with self.assertRaises(TypeError) as error_raised:
+            coreax.util.solve_qp(
+                kernel_mm="invalid_kernel_mm",
```

Comment:
Without the try excepts that have been added in the `coreax.util.solve_qp`, what happens in this case? Does solve_qp throw its own error?

Reply:
The code ultimately returns an error of:

`TypeError: Value 'invalid_kernel_mm' with dtype <U17 is not a valid JAX array type. Only arrays of numeric types are supported by JAX.`

Which is reasonably helpful - the minor modification I made with the try/except logic was just to avoid the enormous error message that proceeds this - and instead highlight only the real issue present